src/ATen/function_wrapper.py

```diff
@@ -918,6 +918,8 @@ def emit_body(env, option):
                 return_tensor = "return Tensor((new ${Tensor}(context,${arg_name}))${maybe_scalar},false);"
                 body.append(CodeTemplate(return_tensor).substitute(
                     env, arg_name=call, maybe_scalar=maybe_scalar))
+            elif ret['type'] == 'accreal' or ret['type'] == 'real':
+                body.append('return scalarTensor({});'.format(call))
```

Comment:
For 'accreal', will this be the right type? For instance, for Byte this will make a ByteTensor, but shouldn't it be a bigger type for accreal?

Reply:
you are correct, I misread the definition of scalarTensor.

Reply:
actually I think we should always return a tensor of the real type for both real and accreal specified returns.

Otherwise consider the following two calls and their respective return types on CPU/GPU for a FloatTensor x (assuming we define real/accreal as it is defined in TH/THC):

| Device | x.sum() |  x.sum(0) |
| ------- | ------- | ---------- |
| CPU     | Double | Float |
| GPU     | Float     | Float |

This is pretty confusing and also inconsistent with what we currently do with the PyTorch Variable API (they always return FloatTensors).

For people worried about precision we should accept a dtype arugment, which we are planning to add to PyTorch as well.